catalog/core/catalog-core-localstorageprovider/src/test/java/org/codice/ddf/catalog/content/impl/FileSystemStorageProviderTest.java

```diff
@@ -244,100 +415,187 @@ public InputStream openStream() throws IOException {
+        String uuid = StringUtils.defaultIfBlank(id,
+                UUID.randomUUID()
+                        .toString()
+                        .replaceAll("-", ""));
         ByteSource byteSource = new ByteSource() {
             @Override
             public InputStream openStream() throws IOException {
                 return IOUtils.toInputStream(data);
             }
         };
-        ContentItem contentItem = new ContentItemImpl(uuid, byteSource, mimeTypeRawData,
+        ContentItem contentItem = new ContentItemImpl(uuid,
+                qualifier,
+                byteSource,
+                mimeTypeRawData,
+                filename,
+                byteSource.size(),
                 mock(Metacard.class));
-        CreateStorageRequest createRequest = new CreateStorageRequestImpl(
-                Collections.singletonList(contentItem), null);
+        CreateStorageRequest createRequest = new CreateStorageRequestImpl(Collections.singletonList(
+                contentItem), null);
         CreateStorageResponse createResponse = provider.create(createRequest);
         List<ContentItem> createdContentItems = createResponse.getCreatedContentItems();
         ContentItem createdContentItem = createdContentItems.get(0);
 
         assertNotNull(createdContentItem);
-        String id = createdContentItem.getId();
-        assertNotNull(id);
-        assertThat(id, equalTo(uuid));
+        String createdId = createdContentItem.getId();
+        assertNotNull(createdId);
+        assertThat(createdId, equalTo(uuid));
 
         String contentUri = createdContentItem.getUri();
         LOGGER.debug("contentUri = {}", contentUri);
         assertNotNull(contentUri);
-        String expectedContentUri = FileSystemStorageProvider.CONTENT_URI_PREFIX + uuid;
+        String expectedContentUri =
+                ContentItem.CONTENT_SCHEME + ":" + uuid + ((StringUtils.isNotBlank(qualifier)) ?
+                        "#" + qualifier :
+                        "");
         assertThat(contentUri, equalTo(expectedContentUri));
 
-        File file = createdContentItem.getFile();
-        assertNotNull(file);
-        assertTrue(file.exists());
         assertTrue(createdContentItem.getSize() > 0);
         String createdMimeType = createdContentItem.getMimeTypeRawData()
                 .replace(";", "");
-        List<String> createdMimeTypeArr = new ArrayList<>(
-                Arrays.asList(createdMimeType.split(" ")));
-        List<String> givenMimeTypeArr = new ArrayList<>(Arrays.asList(
-                mimeTypeRawData.replace(";", "")
-                        .split(" ")));
+        List<String> createdMimeTypeArr =
+                new ArrayList<>(Arrays.asList(createdMimeType.split(" ")));
+        List<String> givenMimeTypeArr = new ArrayList<>(Arrays.asList(mimeTypeRawData.replace(";",
+                "")
+                .split(" ")));
         assertEquals(createdMimeTypeArr.size(), givenMimeTypeArr.size());
```

Comment:
Are these assertions just checking that the lists have the same contents? If so, you could just do `assertThat(createdMimeTypeArr, is(givenMimeTypeArr))`. This would only work if the lists are expected to have the contents in the same order. If they could be in a different order, you could use [`containsInAnyOrder()`](http://hamcrest.org/JavaHamcrest/javadoc/1.3/org/hamcrest/Matchers.html#containsInAnyOrder(T...)).